ConnectionFactory.php

```diff
@@ -45,8 +45,21 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
-            $wrapperClass = null;
+        $allowOverrideUrl = $params['override_url'] ?? false;
+        unset($params['override_url']);
+
+        if (! $allowOverrideUrl) {
+            trigger_deprecation(
+                'doctrine/DoctrineBundle',
+                '2.3.0',
+                'Setting doctrine.dbal.override_url to true allows you to override the dsn parameters set in doctrine.dbal.url. This will be the default behavior in future versions.'
+            );
+        }
+
+        if (! isset($params['pdo']) && (! isset($params['charset']) || $allowOverrideUrl)) {
```

Comment:
the whole `$allowOverrideUrl` should probably be executed only if `$params['url']` is set.

Reply:
good catch & thank you!